src/main/java/org/example/daos/MySqlAuthDao.java

```diff
@@ -0,0 +1,57 @@
+package org.example.daos;
+
+import org.example.daos.interfaces.IAuthDao;
+import org.example.exception.AuthDaoException;
+import org.example.exception.DatabaseConnectionException;
+import org.example.models.LoginRequest;
+import org.example.models.User;
+import org.example.utils.DatabaseConnector;
+import org.mindrot.jbcrypt.BCrypt;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class MySqlAuthDao implements IAuthDao {
+    private DatabaseConnector databaseConnector;
+
+    public MySqlAuthDao(final DatabaseConnector databaseConnector) {
+        this.databaseConnector = databaseConnector;
+    }
+
+    @Override
+    public User getUser(final LoginRequest loginRequest) throws
+            AuthDaoException {
+        User user = new User();
+        String getUserCredQuery =
+                "SELECT `username`, `password`, `salt`,"
+                        + " `sys_role_id` FROM `user`"
+                        + " WHERE `username` = ?";
+        try (Connection connection = databaseConnector.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    getUserCredQuery);
+            statement.setString(1, loginRequest.getUsername());
+            ResultSet resultSet = statement.executeQuery();
+
+            if (!resultSet.next()) {
+                return user;
+            }
+
+            String storedHash = resultSet.getString("password");
+            if (!BCrypt.checkpw(loginRequest.getPassword(), storedHash)) {
+                return user;
```

Comment:
is it better to use java optionals here? https://www.baeldung.com/java-optional so changing the method to return Optional User instead of just User. So that we aren't doing the return user line multiple times. Using optionals would mean we also don't need line 26